zvariant/src/utils.rs

```diff
@@ -316,3 +316,47 @@ where
 {
     input.get(index).ok_or(Error::OutOfBounds)
 }
+
+impl Signature<'_> {
+    /// Determines whether the signature has outer parentheses and returns the
+    /// signature without them if it does.
+    ///    
+    /// # Safety
+    ///
+    /// The byte wise comparison in this method requires the signature to be
+    /// of an encoding that is equal to utf8 with respect to the encoding of
+    /// ASCII characters.
+    /// D-Bus signature type codes each are a single ASCII character.
+    ///
+    pub(crate) fn without_outer_parentheses(&self) -> Option<&[u8]> {
+        let bytes = self.as_bytes();
+
+        if let [b'(', subslice @ .., b')'] = bytes {
```

Comment:
we can get rid of the ugly `b` prefixes and make the code prettier/easier to read if we operate on `char`.

Reply:
I liked the pattern match `[b'(', subslice @ .., b')'] = bytes ..`, as it was likely efficient, avoids indexing and resulted in the subslice in one go.

But I see that handling chars avoids the need for a safety  notice.